ethernet_init/ethernet_init.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-FileCopyrightText: 2022 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Unlicense OR CC0-1.0
+ */
+#pragma once
+
+#include "esp_eth_driver.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Initialize Ethernet driver based on Espressif IoT Development Framework Configuration
+ *
+ * @param[out] eth_handles_out array of initialized Ethernet driver handles
+ * @param[out] eth_cnt_out number of initialized Ethernets
+ * @return
+ *          - ESP_OK on success
+ *          - ESP_ERR_INVALID_ARG when passed invalid pointers
+ *          - ESP_ERR_NO_MEM when there is no memory to allocate for Ethernet driver handles array
+ *          - ESP_FAIL on any other failure
+ */
+esp_err_t ethernet_init(esp_eth_handle_t *eth_handles_out[], uint8_t *eth_cnt_out);
```

Comment:
All the initialization is based on compile time defines, this reflects in not having the selected apparent in the code and the need to check the compile time constant. Given the nature of the device, a PHY that it's likely to change only in hardware revisions, I think it is fine, but we should take this into consideration.

Reply:
rename to ethernet_init_all()
add api to init individual ethernet.

Reply:
I've created a Jira task for adding API to initialise individual ethernet
IDF-8239